### PR TITLE
ref(new-widget-builder-experience): hide sort by step if no group by value is selected - [DD-807]

### PR DESCRIPTION
**Pull request by priscilawebdev**
The SortBy field shall be only visible on tabular visualizations or on time-series _if at least one groupBy value is selected_ 

**What this PR does**

- Move some code around
- Update displaySortBy logic
- Add Test

**Added by Nar**

- fixes the resetting logic when options are selected in the sort by, but removed or changed in their original fields

### CODE REVIEW

**Review by wmak — COMMENTED** [review]
Can we also drop the orderby param from the API request when the group by value isn't selected yet

Should 2 blank groupbys show the orderby?
image
- Probably related, but after removing second group by have to select a column -> delete to hide orderby again

nit:
1. add 1 group by
2. set limit to non default (eg. 2)
3. remove group by
4. add group by
Should the limit from step 2 still be there?

**Comment by priscilawebdev** [comment]
good catch @wmak !! will update the code... thanks

**Comment by narsaynorath** [comment]
@wmak 

I added some code to handle some of the reset/default logic when other fields are changed since it becomes pretty annoying when testing this change. It should also have the following behaviour now:
- Removing the grouping removes the ordering
- Selecting any of the custom equations, y-axes, or groupings to sort by, and making a change to a different field (i.e. ordering by project but changing y-axis) should persist the selection
- When a group by that's selected for ordering is changed, the sort field falls back to the first y-axis
- Sorting by a y-axis and removing that y-axis should fall back to a remaining y-axis (since we don't allow removing if there's one left)

I hope that it's fine for this PR since we're updating some of the logic for displaying the field 🙏 

> Can we also drop the orderby param from the API request when the group by value isn't selected yet
The orderby should be dropped from API requests now

> Should 2 blank groupbys show the orderby?
Nope, Priscila fixed this :)

> Probably related, but after removing second group by have to select a column -> delete to hide orderby again
I think this is fixed now, would you mind confirming?

As for the nit, can we punt on persisting that state? If we've removed the grouping, the user may also have made other changes and I'm not sure if it's a big problem to have them reselect

**Review by wmak — APPROVED** [review]
Issues from earlier comment seem resolved now

edge case (can backlog) but seems related to orderby being blanked out instead of omitted
Add a second group by, don't set the first one and only set the second one. Request will fail cause we're sending a blank orderby param

File: static/app/views/dashboardsV2/widgetBuilder/utils.tsx [hunk 1]
```diff
@@ -123,6 +123,14 @@ export function normalizeQueries({
         query.fields = fields.filter(field => !columns.includes(field));
       }
 
+      if (
```
Comment (shruthilayaj) [inline static/app/views/dashboardsV2/widgetBuilder/utils.tsx:126]: can we add a comment here just so we know why we reset orderby to empty

Comment (narsaynorath) [reply static/app/views/dashboardsV2/widgetBuilder/utils.tsx:126]: https://github.com/getsentry/sentry/pull/33953/commits/a3b19f890a2ad1d2da284b3badfea3ea555d8641
Added! Let me know if it needs improvement

**Review by shruthilayaj — APPROVED** [review]

**Pull request merged**